app/pulp/app/serializers/catalog.py

```diff
@@ -0,0 +1,27 @@
+from rest_framework import serializers
+
+from django.core import validators
+
+from pulp.app import models
+from pulp.app.serializers import ModelSerializer
+
+
+class DownloadCatalogSerializer(ModelSerializer):
+    url = serializers.CharField(
+        help_text="The URL used to download the related artifact.",
+        allow_blank=True, required=False, validators=[validators.URLValidator],
+    )
+
+    artifact = serializers.PrimaryKeyRelatedField(
```

Comment:
Never use PrimaryKeyRelatedField. This should be a HyperlinkedRelatedField referencing the Artifact model.